### DISCUSSION TITLE
new WebSocket(conn) Not connect

### DISCUSSION THREAD

**Discussion by carcleo** [mdn/mdn-community#509] [Content]
Hy my friends

I have it

**index.php**
```
<output></output>
<input type="text">

<script src="assets/js/WebSocket.js"></script>
```

**Soquet.php**

```
<?php

   $host = "127.0.0.1";
   $port = 8080;

   $soquet = socket_create(AF_INET, SOCK_STREAM,SOL_TCP);
   
   socket_bind($soquet, $host, intVal($port));
   socket_listen($soquet);
   socket_accept($soquet);  
   
?>
```

**WebServer.js**

```
const output = document.querySelector("output");
const input = document.querySelector("input");

input.addEventListener("keypress", (e) => {

	if (e.code === "Enter") {

		const port = 8081;
		const host = 'wss://127.0.0.1'; 
		const conn = host + ':' + port;

		const ws = new WebSocket(conn);

		ws.onopen = (e) => {
			console.log(input.value)
			conn.send(input.value);
		};

		ws.onmessage = (e) => {
			console.log(e.data);
			output.value += e.data + "\n";
		};
		
	}
	
})
```

At PHP, i can start the **Socket** normally at the command line (prompt),
but at JS, any port that i put, the reply is always the same:

**WebSocket.js:8 WebSocket connection to 'wss://localhost:8081/' failed:**

Where i am mistaken?

**Comment by bsmth** [comment]
Thanks a lot for posting. For [general coding questions](https://developer.mozilla.org/en-US/docs/MDN/Community#general_support), you can try [Stack Overflow](https://stackoverflow.com/questions/) or other forums that can help you troubleshoot code. I'm going to close this one for now. Thank you :)
